### PR TITLE
Added pythonforandroid.androidndk.AndroidNDK + some changes needed in order to support build on Apple Silicon macs.

### PR DESCRIPTION
**Pull request by misl6**
It started as a minor change to the codebase in order to support builds on Apple Silicon without the aid of Rosetta.

Unfortunately, while doing changes, a lot of issues come up during CI runs (and locally on my M1 Mac), so this PR started to be something bigger, that fixes broken recipes and introduces the new `pythonforandroid.androidndk.AndroidNDK`.

This is a **WIP** PR to track down the current work around `python-for-android` recipes, and is needed in order to launch tests on our self-hosted Apple Silicon runner.

⚠️ Some changes may land into another PR in order to keep track of specific features and / or fixes.

### CODE REVIEW

**Comment by misl6** [comment]
✅ All tests are successful (also the ones on device)
⚠️ We should store all the APKs and AABs, so everything could be tested on a real device (ubuntu + macos-latest + apple-silicon-m1)
✂️ Will start splicing the PR into multiple subsets.

File: pythonforandroid/androidndk.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import sys
+import os
+
+
+class AndroidNDK(object):
```
Comment (AndreMiras) [inline pythonforandroid/androidndk.py:5]: Nit:
```suggestion
class AndroidNDK:
```

File: tests/test_androidndk.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        perform our unittests"""
+        self.ndk = AndroidNDK("/opt/android/android-ndk")
+
+    def test_host_tag(self):
+        """Test the `host_tag` property of the :class:`~pythonforandroid.androidndk.AndroidNDK`
+        class."""
+        if sys.platform == "linux":
+            self.assertEqual(self.ndk.host_tag, "linux-x86_64")
+        elif sys.platform == "darwin":
+            self.assertEqual(self.ndk.host_tag, "darwin-x86_64")
```
Comment (AndreMiras) [inline tests/test_androidndk.py:24]: Why don't we `mock.patch()` the sys.platform ourselves just for the test?

**Review by AndreMiras — APPROVED** [review]
Absolutely awesome you got the Apple Silicon support fully working :clap: 
Thanks a mil!

File: tests/test_androidndk.py (hunk 2 above)
Comment (misl6) [reply tests/test_androidndk.py]: Why not? Addressed 👍

File: pythonforandroid/androidndk.py (hunk 1 above)
Comment (misl6) [reply pythonforandroid/androidndk.py:5]: Guilty of bad old habits (again 😀)

**Review by AndreMiras — APPROVED** [review]
Nice! I can't wait to have this merged

**Pull request merged**